File: examples/governance/operations_domain/lib/operations_domain/incident/events/investigated_incident.rb
module OperationsDomain
  class Incident
    module Events
      class InvestigatedIncident
        attr_reader :aggregate_id, :incident_id, :model_id, :severity, :category, :description, :reported_by_id, :reported_at, :resolved_at, :resolution, :root_cause, :status, :occurred_at

        def initialize(aggregate_id: nil, incident_id: nil, model_id: nil, severity: nil, category: nil, description: nil, reported_by_id: nil, reported_at: nil, resolved_at: nil, resolution: nil, root_cause: nil, status: nil)
          @aggregate_id = aggregate_id
          @incident_id = incident_id
          @model_id = model_id
          @severity = severity
          @category = category
          @description = description
          @reported_by_id = reported_by_id
          @reported_at = reported_at
          @resolved_at = resolved_at
          @resolution = resolution
          @root_cause = root_cause
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end

File: examples/governance_static_go/domain/investigated_incident.go
package domain

import "time"

type InvestigatedIncident struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	IncidentId string `json:"incident_id"`
	ModelId string `json:"model_id"`
	Severity string `json:"severity"`
	Category string `json:"category"`
	Description string `json:"description"`
	ReportedById string `json:"reported_by_id"`
	ReportedAt time.Time `json:"reported_at"`
	ResolvedAt time.Time `json:"resolved_at"`
	Resolution string `json:"resolution"`
	RootCause string `json:"root_cause"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e InvestigatedIncident) EventName() string { return "InvestigatedIncident" }

func (e InvestigatedIncident) GetOccurredAt() time.Time { return e.OccurredAt }
